plugins/woocommerce/includes/abstracts/abstract-wc-data.php

```diff
@@ -485,11 +485,11 @@ public function update_meta_data( $key, $value, $meta_id = 0 ) {
 			}
 
 			if ( ! empty( $matches ) ) {
-				// Set matches to null so only one key gets the new value.
+				// Update first match and delete the rest.
+				$array_key = array_shift( $matches );
```

Comment:
This was a core bug which caused duplicate keys when using `update_meta_data`. We were using `add_meta_data` as a workaround. This fixes it.